tests/stub/routing.py

```diff
@@ -2031,11 +2296,11 @@ def test_should_request_rt_from_all_initial_routers_until_successful(self):
 
         # The resolver is used to convert the original address to multiple fake domain names.
         def resolver(address):
-            nonlocal resolver_call_num
+            nonlocal resolver_calls
             nonlocal resolved_addresses
-            self.assertEqual(0, resolver_call_num)
-            self.assertEqual(self._routingServer1.address, address)
-            resolver_call_num += 1
+            resolver_calls[address] = resolver_calls.get(address, 0) + 1
+            if address != self._routingServer1.address:
```

Comment:
What do we need this `if` statement for?

Reply:
The Python driver calls the custom resolver function for the initial address as well as for each address that is received as part of a routing table ([Trello card](https://trello.com/c/HhizyhRm/3495)). This behavior will not be unified before 5.0 because it's a breaking change.